src/Control/Monad/Free.purs

```diff
@@ -41,6 +41,19 @@ data FreeView f a b = Return a | Bind (f b) (b -> Free f a)
 
 data Val
 
+instance eqFree :: (Functor f, Eq (f (Free f a)), Eq a) => Eq (Free f a) where
+  eq x y = case resume x, resume y of
+    Left fa, Left fb -> fa == fb
+    Right a, Right b -> a == b
+    _, _ -> false
```

Comment:
Yeah, good point. Same with compare too! I wrote these with an `Eq1` first, and the cases are a little different in that case.